docs/source/cheat_sheet_py3.rst

```diff
@@ -105,6 +105,45 @@ Python 3 introduces an annotation syntax for function declarations in `PEP 3107
        
        ...
 
+Coroutines and asyncio
+**********************
+
+See :ref:`async-and-await` for the full detail on typing coroutines and asynchronous code.
+
+.. code-block:: python
+
+   import asyncio
+   from typing import Generator, Any
+
+   # A Python 3.4 coroutine should have a return type of
+   # Generator[Any, None, T], where T is the type it returns.
+   @asyncio.coroutine
+   def countdown34(tag: str, count: int) -> Generator[Any, None, str]:
+       while count > 0:
+           print('T-minus {} ({})'.format(count, tag))
+           yield from asyncio.sleep(0.1)
+           count -= 1
+       return "Blastoff!"
+
+   # mypy currently does not support converting functions into coroutines in
+   # Python 3.4, so you need to add a 'yield' to make it typecheck.
```

Comment:
I think it is important to say "dynamically convert a function into an async generator" as mypy *can* infer explicit generators are generators.